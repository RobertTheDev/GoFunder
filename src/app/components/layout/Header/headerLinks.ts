const headerLinks: {
    href: string;
    name: string;
}[] = [
    { href: '/', name: 'Home' },
    { href: '/fundraisers', name: 'Explore' },
    { href: '/create-fundraiser', name: 'Start Fundraising' }
];

export default headerLinks;
