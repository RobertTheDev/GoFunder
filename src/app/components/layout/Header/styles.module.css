.headerContainer {
    align-items: center;
    background-color: white;
    box-shadow: 0 4px 6px -1px rgb(0 0 0 / 10%), 0 2px 4px -2px rgb(0 0 0 / 10%);
    display: flex;
    height: 6.4rem;
    padding: 0 var(--spacer-lg1);
    position: sticky;
    top: 0;
    width: 100%;
    z-index: 22;
}

.headerLogo {
    color: black;
    font-size: 1.8rem;
    font-weight: 600;
    text-decoration: none;
}