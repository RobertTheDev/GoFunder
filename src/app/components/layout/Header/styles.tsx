import Link from 'next/link';
import styled from 'styled-components';

export const StyledHeaderContainer = styled.header`
    align-items: center;
    background-color: white;
    box-shadow:
        0 4px 6px -1px rgb(0 0 0 / 10%),
        0 2px 4px -2px rgb(0 0 0 / 10%);
    display: flex;
    height: 6.4rem;
    padding: 0 var(--spacer-lg1);
    position: sticky;
    top: 0;
    width: 100%;
    z-index: 22;
`;

export const StyledHeaderLogo = styled(Link)`
    color: black;
    font-size: 1.8rem;
    font-weight: 600;
    text-decoration: none;
`;
