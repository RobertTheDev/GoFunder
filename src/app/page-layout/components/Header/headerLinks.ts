const headerLinks = [
    { href: "/", name: "Home" },
    { href: "/fundraisers", name: "Explore" },
    { href: "/fundraisers/create", name: "Start Fundraising" },
];

export default headerLinks;
