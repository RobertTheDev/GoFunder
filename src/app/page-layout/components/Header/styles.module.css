.headerContainer {
    align-items: center;
    background-color: white;
    box-shadow: var(--shadow-md);
    display: flex;
    justify-content: space-between;
    padding: 1.8rem 2.4rem;
    position: sticky;
    top: 0;
    width: 100%;
    z-index: 22;
    box-sizing: border-box;
    overflow: hidden;
    max-height: 8rem;

    @media (width >=1024px) {
        padding: 2.4rem 8rem;
    }

    @media (prefers-color-scheme: dark) {
        background-color: var(--slate-950);
    }
}

.headerLinkText {
    font-size: 1.6rem;
    font-weight: 400;
    text-decoration: none;
    transition: 0.2s ease-in-out;

    @media (prefers-color-scheme: dark) {
        color: var(--slate-200);
    }
}

.headerLinkText:hover {
    color: var(--primary600);
}

.headerLinkActive {
    color: var(--primary600);
    font-size: 1.6rem;
    font-weight: 400;
    text-decoration: none;
}

.headerLinkMenu {
    align-items: center;
    display: none;
    gap: 3.2rem;

    @media (width >=1024px) {
        display: flex;
    }
}

.headerLogo {
    color: var(--green-600);
    font-size: 2.2rem;
    font-weight: 600;
    text-decoration: none;
}

.headerAvatarInput {
    border-radius: 2.4rem;
    height: 4.4rem;
    object-fit: cover;
    width: 4.4rem;

}

.headerContentContainer {
    align-items: center;
    display: flex;
    gap: 3.2rem;
}



.headerSignUpButton {
    background-color: var(--primary600);
    color: white;
    border-radius: 10rem;
    font-size: 1.6rem;
    font-weight: 400;
    padding: 1.8rem 3.2rem;
    transition: 0.4s ease-in-out;
}

.headerSignUpButton:hover {
    background-color: var(--primary700);
}

.headerControlsContainer {
    align-items: center;
    display: none;
    gap: 0.8rem;

    @media (width >=1024px) {
        display: flex;
    }
}