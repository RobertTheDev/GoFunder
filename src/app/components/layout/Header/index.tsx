'use client';

import Link from 'next/link';
import ProfileMenu from '../../profile/ProfileMenu';
import { StyledHeaderContainer, StyledHeaderLogo } from './styles';

export default function Header() {
    const headerLinks: {
        href: string;
        name: string;
    }[] = [
        {
            href: '/about',
            name: 'About'
        },
        { href: '/', name: 'Find Fundraisers' },
        { href: '/create-fundraiser', name: 'Create A Fundraiser' }
    ];

    return (
        <StyledHeaderContainer>
            <StyledHeaderLogo href="/">GoFunder</StyledHeaderLogo>
            <nav>
                {headerLinks.map(({ href, name }) => (
                    <Link key={name} href={href}>
                        {name}
                    </Link>
                ))}
            </nav>
            <div>
                <Link href="/saved-fundraisers">Saved Fundraisers</Link>
                <button type="button">Profile</button>
                <ProfileMenu />
            </div>
        </StyledHeaderContainer>
    );
}
