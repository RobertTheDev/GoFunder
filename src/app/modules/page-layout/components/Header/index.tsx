'use client';

import Link from 'next/link';
import { useProfileMenu } from '@/app/hooks/profileMenu/useProfileMenu';
import { usePathname, useRouter } from 'next/navigation';
import ProfileMenu from '../../../profile/components/ProfileMenu';
import headerLinks from './headerLinks';
import styles from './styles.module.css';

export default function Header() {
    const { toggleProfileMenu, profileMenuActive, profileMenuRef } =
        useProfileMenu();

    const router = useRouter();

    const pathName = usePathname();

    return (
        <header className={styles.headerContainer}>
            <div className={styles.headerContentContainer}>
                <Link href="/" className={styles.headerLogo}>
                    GoFunder
                </Link>
                <nav className={styles.headerLinkMenu}>
                    {headerLinks.map(({ href, name }) => (
                        <Link
                            key={name}
                            href={href}
                            className={
                                pathName === href
                                    ? styles.headerLinkActive
                                    : styles.headerLinkText
                            }
                        >
                            {name}
                        </Link>
                    ))}
                </nav>
            </div>
            <div className={styles.headerControlsContainer}>
                <button
                    className={styles.headerSignInButton}
                    type="button"
                    onClick={() => router.push('/auth/sign-in')}
                >
                    Sign In
                </button>
                <button
                    className={styles.headerSignUpButton}
                    type="button"
                    onClick={() => router.push('/auth/sign-up')}
                >
                    Sign Up
                </button>

                <input
                    type="image"
                    src="https://images.unsplash.com/photo-1564564321837-a57b7070ac4f?q=80&w=2952&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
                    alt="avatar image"
                    className={styles.headerAvatarInput}
                    onClick={() => toggleProfileMenu()}
                />
                {profileMenuActive && <ProfileMenu ref={profileMenuRef} />}
            </div>
        </header>
    );
}
