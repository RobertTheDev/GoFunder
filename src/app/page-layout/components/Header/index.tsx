"use client";

import Link from "next/link";
import useProfileMenu from "@/app/hooks/profileMenu/useProfileMenu";
import { usePathname } from "next/navigation";
import headerLinks from "./headerLinks";
import styles from "./styles.module.css";
import ProfileMenu from "../ProfileMenu";

export default function Header() {
    const { toggleProfileMenu, profileMenuActive, profileMenuRef } =
        useProfileMenu();

    const pathName = usePathname();

    return (
        <header className={styles.headerContainer}>
            <div className={styles.headerContentContainer}>
                <Link href="/" className={styles.headerLogo}>
                    GoFunder
                </Link>
                <nav className={styles.headerLinkMenu}>
                    {headerLinks.map(({ href, name }) => (
                        <Link
                            key={name}
                            href={href}
                            className={
                                pathName === href
                                    ? styles.headerLinkActive
                                    : styles.headerLinkText
                            }
                        >
                            {name}
                        </Link>
                    ))}
                </nav>
            </div>
            <div className={styles.headerControlsContainer}>
                {/* <Link href="/auth/sign-in" className={styles.headerLinkText}>
                    Sign In
                </Link> */}

                <input
                    type="image"
                    src="https://images.unsplash.com/photo-1564564321837-a57b7070ac4f?q=80&w=2952&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
                    alt="avatar image"
                    className={styles.headerAvatarInput}
                    onClick={() => toggleProfileMenu()}
                />
                {profileMenuActive && <ProfileMenu ref={profileMenuRef} />}
            </div>
        </header>
    );
}
