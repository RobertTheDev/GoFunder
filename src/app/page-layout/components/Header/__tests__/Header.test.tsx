import { describe, it, expect, vi } from "vitest";
import { render } from "@testing-library/react";
import Header from "../index";

vi.mock("next/navigation", () => ({
    usePathname: vi.fn(() => "/"),
    useRouter: vi.fn(() => ({ pathname: "/" })),
}));

describe("Header component", () => {
    it("renders without crashing", () => {
        render(<Header />);
    });

    it("matches snapshot", () => {
        const { asFragment } = render(<Header />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("displays correct number of links", () => {
        const { getAllByRole } = render(<Header />);
        const links = getAllByRole("link");
        expect(links).toHaveLength(4);
    });

    it("hides profile menu by default", () => {
        const { queryByTestId } = render(<Header />);
        const profileMenu = queryByTestId("profile-menu");
        expect(profileMenu).toBeNull();
    });
});
