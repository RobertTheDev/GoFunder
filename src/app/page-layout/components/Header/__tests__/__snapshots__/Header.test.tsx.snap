// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="_headerContainer_872daa"
  >
    <div
      class="_headerContentContainer_872daa"
    >
      <a
        class="_headerLogo_872daa"
        href="/"
      >
        GoFunder
      </a>
      <nav
        class="_headerLinkMenu_872daa"
      >
        <a
          class="_headerLinkActive_872daa"
          href="/"
        >
          Home
        </a>
        <a
          class="_headerLinkText_872daa"
          href="/fundraisers"
        >
          Explore
        </a>
        <a
          class="_headerLinkText_872daa"
          href="/fundraisers/create"
        >
          Start Fundraising
        </a>
      </nav>
    </div>
    <div
      class="_headerControlsContainer_872daa"
    >
      <input
        alt="avatar image"
        class="_headerAvatarInput_872daa"
        src="https://images.unsplash.com/photo-1564564321837-a57b7070ac4f?q=80&w=2952&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
        type="image"
      />
    </div>
  </header>
</DocumentFragment>
`;
